ksql/src/main/java/org/apache/zeppelin/ksql/KSQLInterpreter.java

```diff
@@ -0,0 +1,170 @@
+/*
+* Copyright 2016 Google Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package org.apache.zeppelin.ksql;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang.StringUtils;
+import org.apache.zeppelin.interpreter.Interpreter;
+import org.apache.zeppelin.interpreter.InterpreterContext;
+import org.apache.zeppelin.interpreter.InterpreterException;
+import org.apache.zeppelin.interpreter.InterpreterResult;
+import org.apache.zeppelin.interpreter.util.InterpreterOutputStream;
+import org.apache.zeppelin.scheduler.Scheduler;
+import org.apache.zeppelin.scheduler.SchedulerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+
+
+public class KSQLInterpreter extends Interpreter {
+  private static final String NEW_LINE = "\n";
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(KSQLInterpreter.class);
+  public static final String TABLE_DELIMITER = "\t";
+
+  private InterpreterOutputStream interpreterOutput = new InterpreterOutputStream(LOGGER);
+
+  private final KSQLRestService ksqlRestService;
+
+  private static final ObjectMapper json = new ObjectMapper();
+
+  public KSQLInterpreter(Properties properties) {
+    this(properties, new KSQLRestService(properties.entrySet().stream()
+            .collect(Collectors.toMap(e -> e.getKey().toString(),
+                e -> e.getValue() != null ? e.getValue().toString() : null))));
+  }
+
+  // VisibleForTesting
+  public KSQLInterpreter(Properties properties, KSQLRestService ksqlRestService) {
+    super(properties);
+    this.ksqlRestService = ksqlRestService;
+  }
+
+  @Override
+  public void open() throws InterpreterException {}
```

Comment:
It is better to create and initialize `ksqlRestService` in `open` method.

Reply:
Can we left it as is? It's better for testing so I can inject the mocked version, and the behaviour si lazy for each paragraph

Reply:
OK